client/containers/Pub/pubHooks.ts

```diff
@@ -19,3 +22,43 @@ export const usePubData = () => {
 	const { pubData } = useContext(PubContext);
 	return pubData;
 };
+
+const getRenderedValue = (note) => {
+	const hasRenderedStructuredValue = note.renderedStructuredValue?.json?.length;
+	return hasRenderedStructuredValue ? note.renderedStructuredValue.html : note.unstructuredValue;
+};
+
+export const usePubNotes = () => {
+	const {
+		noteManager,
+		collabData: {
+			editorChangeObject: { view },
+		},
+	} = usePubContext();
+	const { citations = [], footnotes = [] } = view ? getNotes(view.state.doc) : {};
+	const [notes, setNotes] = useState(noteManager.notes);
+	useEffect(() => noteManager.subscribe(setNotes), [noteManager]);
```

Comment:
$0.02: I like to write `useEffect` calls after all of the variable declarations (except handlers and render helpers) inside of a component or hook, roughly in this order.

```
const Component = (props) => {
    const {x, y, z} = props
    const [a, setA] = useState(...)
    const [b, setB] = useState(...)
    const {q, r, s} = useContext(...)
    const w = fn(x, y, z);

    useEffect(...)
    useEffect(...)

    const handleX = useCallback(() => ...)

    const renderX = () => ...
    const renderY = () => ...

    return ...
}
```

So it ends up roooooughly in this order:

1. Variable declarations (except nested functions) including those that call hooks
2. Effect hooks that don't return a value (e.g. `useEffect`, our `useInfiniteScroll`)
3. Event handler declarations (e.g. `handleClick`)
4. Render helpers (e.g. `renderInnerText`)
5. The return value

You're not required to use this ordering, but I'm curious if you have any of your own headcanon conventions for how these things should be laid out. For instance, it makes sense that you'd have the `useState` and `useEffect` hooks related to `notes` in the same place — it's just not where I'd expect to find that `useEffect`.